osu.Game/Online/Multiplayer/MultiplayerClient.cs

```diff
@@ -140,6 +139,9 @@ public async Task JoinRoom(Room room, string? password = null)
                 var joinedRoom = await JoinRoom(room.RoomID.Value.Value, password ?? room.Password.Value).ConfigureAwait(false);
                 Debug.Assert(joinedRoom != null);
 
+                // Populate playlist items.
+                var playlistItems = await Task.WhenAll(joinedRoom.Playlist.Select(createPlaylistItem)).ConfigureAwait(false);
```

Comment:
Not too confident on this one. If a user joins a room with a large history is this going to potentially take seconds or minutes? Maybe okay for an initial release, but probably not as a permanent solution. The population of everything that isn't the beatmap id should likely happen separate from the join room operation.

Reply:
Sure, but that's more involved to do. Too out of scope for this PR, imo.

Reply:
of course